libbeat/processors/add_kubernetes_metadata/_meta/fields.yml

```diff
@@ -28,6 +28,11 @@
           description: >
             Kubernetes node name
 
+        - name: node.hostname
+          type: keyword
+          description: >
+            Kubernetes hostname as reported by the node’s kernel
```

Comment:
Thinking about this comment https://github.com/elastic/beats/issues/20434#issuecomment-720382877, perhaps we should store this value directly in `host.name` (instead of storing it in `kubernetes.node.hostname`). What I am not sure is if then we should also populate `host.id` or other inventory fields and with what values.

Other option can be to postpone the decision and copy the field if we decide to use `host.name`.

@exekias @kaiyan-sheng wdyt?

Reply:
Using `host.name` sounds like a good idea to me, this complies with the current definition we use. It would be good to do some testing to see if what `add_host_metadata` reports is the same thing reported by Kubernetes. For instance, when you change the hostname by hand, with and without setting a fqdn.

I think we need a deeper discussion around where `host.id` should be set, what will happen when `add_host_metadata` kicks in but `host.name` is already set by `add_kubernetes_metadata`? What will happen if `add_host_metadata`  happens before `add_kubernetes_metadata`?

Reply:
That sounds good to me! How about keeping the scope of this PR to just adding the node's metadata and opening a follow up issue so as to investigate further about `host.name` in separate story?

Reply:
> How about keeping the scope of this PR to just adding the node's metadata and opening a follow up issue so as to investigate further about host.name in separate story?

LGTM, worst case we will duplicate this field.

Reply:
> I think we need a deeper discussion around where `host.id` should be set, what will happen when `add_host_metadata` kicks in but `host.name` is already set by `add_kubernetes_metadata`? What will happen if `add_host_metadata` happens before `add_kubernetes_metadata`?

Agree. This is definitely worth some time to test and discuss. When users have more than one processor enabled for Metricbeat, overwriting the same fields will be a problem. We also have this[ issue](https://github.com/elastic/beats/issues/21773) to discuss moving adding `host.name` into `add_host_metadata`. But then both `host.id` and `host.name` will have this same issue.